src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.BatchingDocumentCollection.cs

```diff
@@ -537,40 +543,63 @@ public void ReorderFiles(ImmutableArray<string> filePaths)
             Func<Solution, ImmutableArray<DocumentId>, Solution> removeDocuments,
             WorkspaceChangeKind removeDocumentChangeKind)
         {
-            // Document adding...
-            solutionChanges.UpdateSolutionForDocumentAction(
-                newSolution: addDocuments(solutionChanges.Solution, _documentsAddedInBatch.ToImmutable()),
-                changeKind: addDocumentChangeKind,
-                documentIds: _documentsAddedInBatch.Select(d => d.Id));
-
-            foreach (var documentInfo in _documentsAddedInBatch)
+            UpdateSolutionForBatch(solutionChanges, documentFileNamesAdded, documentsToOpen, addDocuments,
+                addDocumentChangeKind, removeDocuments, removeDocumentChangeKind, _project.Id, _documentsAddedInBatch.ToImmutableArray(),
+                _documentsRemovedInBatch.ToImmutableArray(), _orderedDocumentsInBatch,
```

Comment:
i would put a comment here that you're intentionally making a copy, and what ends up actually clearing this once we succeed at updating the solution.